src/aind_data_schema/utils/validators.py

```diff
@@ -106,3 +108,26 @@ def recursive_get_all_names(obj: Any) -> List[str]:
             names.extend(recursive_get_all_names(field_value))
 
     return names
+
+
+def recursive_check_paths(obj: Any, directory: Optional[Path] = None):
+    """
+    Recursively checks if any attribute in an object is a pathlib.Path and logs if it doesn't exist.
+    """
+    if isinstance(obj, Enum):
+        return
+
+    if isinstance(obj, PurePosixPath):
+        obj = Path(obj)
+
+    if isinstance(obj, Path):
+        full_path = directory / obj if directory else obj
+        if not full_path.exists():
+            logging.warning(f"AssetPath {full_path} does not exist")
```

Comment:
maybe add soft validation of the relative path aspect too, something like
```suggestion
            logging.warning(f"AssetPath {full_path} does not exist")
        if obj.is_absolute():
            try:
                relative_path = obj.relative_to(directory)
            except ValueError:
                logging.warning(f"AssetPath {obj} is absolute, should be relative")
            else:
                logging.warning(f"AssetPath {obj} is absolute, replace with {relative_path}")
    ```

Reply:
(don't think its possible to actually change the path object in-place)

Reply:
This should probably be a validator, I'll add it as one

Reply:
Oh wait right these aren't pydantic models... I guess this is another recursive validator?

Reply:
we could do it as a validator i think by adding it to your __get_pydantic_core_schema__ call, just would lose the ability to check relative to the write directory and suggest a replacement. that's probably fine, although i suppose a separate recursive validator could actually make the replacement if structured differently. don't think i have an opinion.

Reply:
I'm just adding it to the recursive checker. Good enough